numpy/lib/_iotools.py

```diff
@@ -529,7 +541,8 @@ class StringConverter(object):
     _mapper.extend([(nx.floating, float, nx.nan),
                     (complex, _bytes_to_complex, nx.nan + 0j),
                     (nx.longdouble, nx.longdouble, nx.nan),
-                    (nx.string_, bytes, b'???')])
+                    (nx.unicode_, asunicode, '???'),
+                    (nx.string_, asbytes, '???')])
```

Comment:
I don't like it either, but I have not succeeded in creating a testcase that fails here yet.